packages/xstate-react/src/createActorContext.ts

```diff
@@ -67,20 +80,22 @@ export function createActorContext<TMachine extends AnyStateMachine>(
     machine: providedMachine = machine
   }: {
     children: React.ReactNode;
-    machine: TMachine;
+    machine: TBehavior;
   }) {
     const actor = (useActorRef as any)(
       providedMachine,
       interpreterOptions,
       observerOrListener
-    ) as ActorRefFrom<TMachine>;
+    ) as ActorRefFrom<TBehavior>;
 
     return React.createElement(OriginalProvider, { value: actor, children });
   }
 
-  Provider.displayName = `ActorProvider(${machine.id})`;
+  Provider.displayName = `ActorProvider(${
+    (machine as unknown as AnyStateMachine).id ?? machine.transition.name
```

Comment:
naughty

Reply:
I'm not sure if `?? machine.transition.name` is really that great here, that usually would result in just `'transition'`